cmd/exec/operations.go

```diff
@@ -527,11 +527,12 @@ func executeEvent(url string) (string, error) {
 		}
 
 		if resp.StatusCode != http.StatusOK {
-			return "", errors.New("Bad request")
+			log.Fatalf("Failed to send event: %v", err)
 		}
 	}
 
-	if event["id"] != nil {
+	_, ok := event["id"].(string)
```

Comment:
You actually need two checks here. The first, which is missing, is whether `id` is defined in the map at all. The second is the one you're checking here: if the value (assuming it exists) is a string.

```
x, exists := event["id"]
if !exists {
  // do something
}

id, ok := x.(string)
if !ok {
  // do something else
}

// all good
```